src/envoy/http/jwt_auth/jwt_authenticator.cc

```diff
@@ -205,11 +195,11 @@ void JwtAuthenticator::VerifyKey(const PubkeyCacheItem& issuer_item) {
     return;
   }
 
-  if (!issuer_item.jwt_config().forward_payload_header().empty()) {
-    const LowerCaseString key(
-        issuer_item.jwt_config().forward_payload_header());
-    headers_->addCopy(key, jwt_->PayloadStrBase64Url());
-  }
+  // TODO: can we save as proto or json object directly?
+  // User the issuer as the entry key for simplicity. The forward_payload_header
+  // field can be removed or replace by a boolean (to make `save` is
+  // conditional)
```

Comment:
We may have to do both.   I am afraid some backend app may expect payload if "forward_payload_header" is set.

Reply:
We didn't (want) expose forward_payload_header to user. It might be the case for the use of JWT filter alone, but not for the authn flow in Istio. Currently, these outputs are removed by authn filter, not to mention the header key (assigned by pilot) is a hash of issuer, so fairly unusable. On the other hand, they can always use the original JWT anyway.